plugin.video.science.friday/resources/lib/main.py

```diff
@@ -0,0 +1,133 @@
+def root(plugin):
+    """:type plugin: Route"""
+    # Set context parameters based on default view setting
+    if plugin.setting.get_int("defaultview") == 0:
+        context_label = plugin.localize(LIST_AUDIO)
+        context_type = "segment"
+        item_type = "video"
+    else:
+        context_label = plugin.localize(LIST_VIDEO)
+        context_type = "video"
+        item_type = "segment"
+
+    # Fetch HTML Source
+    url = "https://www.sciencefriday.com/explore/"
+    html = urlquick.get(url)
+
+    # Parse for the content
+    root_elem = html.parse("form", attrs={"class": "searchandfilter"})
+    sfid = root_elem.get("data-sf-form-id")
+
+    # Add Youtube & Recent Content
+    yield Listitem.youtube("UCDjGU4DP3b-eGxrsipCvoVQ")
+
+    # Add Recent Videos link
+    yield Listitem.from_dict(content_lister, bold(plugin.localize(RECENT_VIDEOS)),
+                             params={"sfid": sfid, "ctype": "video"})
+    # Add Recent Audio link
+    yield Listitem.from_dict(content_lister, bold(plugin.localize(RECENT_AUDIO)),
+                             params={"sfid": sfid, "ctype": "segment"})
+
+    # List all topics
+    for elem in root_elem.iterfind(".//option[@data-sf-cr]"):
+        item = Listitem()
+        item.label = elem.text
+
+        # Add context item to link to the opposite content type. e.g. audio if video is default
+        item.context.container(content_lister, context_label, topic=elem.attrib["value"], sfid=sfid, ctype=context_type)
+        item.set_callback(content_lister, topic=elem.attrib["value"], ctype=item_type, sfid=sfid)
+        yield item
+
+
+@Route.register
+def content_lister(plugin, sfid, ctype, topic=None, page_count=1):
+    """
+    :type plugin: Route
+    :type sfid: unicode
+    :type ctype: unicode
+    :type topic: unicode
+    :type page_count: int
+    """
+    # Add link to Alternitve Listing
+    if page_count == 1 and topic:
+        params = {"_updatelisting_": True, "sfid": sfid, "topic": topic,
+                  "ctype": u"segment" if ctype == u"video" else u"video"}
+        label = bold(plugin.localize(LIST_AUDIO) if ctype == u"video" else plugin.localize(LIST_VIDEO))
+        item_dict = {"label": label, "callback": content_lister, "params": params}
+        yield Listitem.from_dict(**item_dict)
+
+    # Create content url
+    if topic:
+        url = "https://www.sciencefriday.com/wp-admin/admin-ajax.php?action=get_results&paged=%(next)s&" \
+              "sfid=%(sfid)s&post_types=%(ctype)s&_sft_topic=%(topic)s" % \
+              {"sfid": sfid, "ctype": ctype, "topic": topic, "next": page_count}
+    else:
+        url = "https://www.sciencefriday.com/wp-admin/admin-ajax.php?action=get_results&paged=%(next)s&" \
+              "sfid=%(sfid)s&post_types=%(ctype)s" % \
+              {"sfid": sfid, "ctype": ctype, "next": page_count}
+
+    # Fetch & parse HTML Source
+    ishd = bool(plugin.setting.get_int("video_quality", addon_id="script.module.youtube.dl"))
```

Comment:
`script.module.youtube.dl` is missing in the addon.xml, but code looks like it's not optional.